vendors/vendor/boards/board/ports/ota_pal_for_aws/ota_pal.h

```diff
@@ -0,0 +1,231 @@
+/*
+ * FreeRTOS V202012.00
+ * Copyright (C) 2019 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/**
+ * @file  ota_pal.h
+ * @brief Function declarations for the functions in ota_pal.c
+ */
+
+#ifndef OTA_PAL_H_
+#define OTA_PAL_H_
+
+#include "ota.h"
+
+/**
+ * @brief Abort an OTA transfer.
+ *
+ * Aborts access to an existing open file represented by the OTA file context pFileContext. This is
+ * only valid for jobs that started successfully.
+ *
+ * @note The input OtaFileContext_t pFileContext is checked for NULL by the OTA agent before this
+ * function is called.
+ *
+ * This function may be called before the file is opened, so the file pointer pFileContext->fileHandle
+ * may be NULL when this function is called.
+ *
+ * The file pointer will be set to NULL after this function returns.
+ *
+ * @param[in] pFileContext OTA file context information.
+ *
+ * @return The OtaPalStatus_t error code is a combination of the main OTA PAL interface error and
+ *         the MCU specific sub error code. See ota_platform_interface.h for the OtaPalMainStatus_t
+ *         error codes and your specific PAL implementation for the sub error code.
+ *
+ * Major error codes returned are:
+ *
+ *   OtaPalSuccess: Aborting access to the open file was successful.
+ *   OtaPalFileAbort: Aborting access to the open file context was unsuccessful.
+ */
+OtaPalStatus_t otaPal_Abort( OtaFileContext_t * const pFileContext );
+
+/**
+ * @brief Create a new receive file for the data chunks as they come in.
+ *
+ * @note Opens the file indicated in the OTA file context in the MCU file system.
```

Comment:
Not all ports will have file system so also mention about the flash

Reply:
Can you recommend a sentence for this? I'm not as familiar with the flash devices so I'm not sure what wording would be appropriate.